src/agent/block_store_services/block_store_client.js

```diff
@@ -1,20 +1,28 @@
 /* Copyright (C) 2016 NooBaa */
 'use strict';
 
-const request = require('request');
 const xml2js = require('xml2js');
 const AWS = require('aws-sdk');
 const _ = require('lodash');
 
 const azure_storage = require('../../util/azure_storage_wrap');
 const P = require('../../util/promise');
+const promise_utils = require('../../util/promise_utils');
 const dbg = require('../../util/debug_module')(__filename);
 const buffer_utils = require('../../util/buffer_utils');
 const http_utils = require('../../util/http_utils');
+const LRUCache = require('../../util/lru_cache');
 const config = require('../../../config');
-const { RPC_BUFFERS } = require('../../rpc');
-
-
+const { RPC_BUFFERS, RpcError } = require('../../rpc');
+const { get_block_internal_dir } = require('../../agent/block_store_services/block_store_base');
+
+const block_store_info_cache = new LRUCache({
+    name: 'BlockStoreInfoCache',
+    max_usage: 1000,
+    expiry_ms: 2 * 60 * 1000,
```

Comment:
I think we can allow more than 2 minutes before expiry. maybe 10? ✋🤚

Reply:
kept on 2... we'll see in the future